### PR TITLE
Feat(addon): add more kustomize traits

### PR DESCRIPTION
**Pull request by FogDong**
**What this PR does / why we need it**:
add more kustomize traits

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
/cc @wonderflow @chivalryq

### CODE REVIEW

File: vela-templates/addons/fluxcd/definitions/kustomize-images.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      template: |
+        patch: {
+        	spec: {
+        		images: parameter.images
+        	}
+        }
+
+        parameter: {
+        	// +usage=specify the target namespace of kustomize
+        	targetNamespace: string
```
Comment (chivalryq) [inline vela-templates/addons/fluxcd/definitions/kustomize-images.yaml:19]: should be image

File: vela-templates/addons/fluxcd/definitions/target-namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        parameter: {
+        	// +usage=specify the images of kustomize
+        	images: [...{
+            name: string
+            newName?: string
+            newTag?: string
+            digest?: string
+          }]
```
Comment (chivalryq) [inline vela-templates/addons/fluxcd/definitions/target-namespace.yaml:24]: Alright...This seems to be targetNamespace

**Review by chivalryq — CHANGES_REQUESTED** [review]

File: vela-templates/addons/fluxcd/definitions/kustomize-images.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        parameter: {
+        	// +usage=specify the images of kustomize
+        	images: [...{
+            name: string
+            newName?: string
+            newTag?: string
+            digest?: string
+          }]
+        }
```
Comment (wonderflow) [inline vela-templates/addons/fluxcd/definitions/kustomize-images.yaml:25]: why it's a trait? This should be component properties

**Comment by FogDong** [comment]
We don't need these traits now since we have `envbinding` to complete these features.

/close

**Pull request closed without merging**